src/import_vdf/milvus_import.py

```diff
@@ -0,0 +1,115 @@
+import os
+from dotenv import load_dotenv
+import pandas as pd
+from tqdm import tqdm
+
+from pymilvus import connections, utility, Collection, CollectionSchema, FieldSchema, DataType
+
+from names import DBNames
+from util import standardize_metric_reverse
+from import_vdf.vdf_import_cls import ImportVDF
+
+
+load_dotenv()
+
+
+class ImportMilvus(ImportVDF):
+    DB_NAME_SLUG = DBNames.MILVUS
+
+    def __init__(self, args):
+        # call super class constructor
+        super().__init__(args)
+        if args is None:
+            args = {}
+        assert isinstance(args, dict), 'Invalid args.'
+        super().__init__(args)
+        
+        uri = self.args.get('uri', 'http://localhost:19530')
+        token = self.args.get('token', '')
+        connections.connect(uri=uri, token=token)
+
+    def upsert_data(self):
+        # we know that the self.vdf_meta["indexes"] is a list
+        for collection_name, index_meta in self.vdf_meta['indexes'].items():
+            # load data
+            print(f'Importing data for collection "{collection_name}"')
+            for namespace_meta in index_meta:
+                data_path = namespace_meta['data_path']
+                index_name = namespace_meta['index_name']
+
+                # check if collection exists
+                if utility.has_collection(collection_name):
+                    collection = Collection(collection_name)
+                    f_vector = None
+                    f_pk = None
+                    for f in collection.schema.fields:
+                        if f.dtype.value in [100, 101]:
+                            f_vector = f
+                        if hasattr(f, 'is_primary') and f.is_primary:
+                            f_pk = f
+                else:
+                    # create collection
+                    try:
+                        f_pk = FieldSchema(
+                            name='pk', dtype=DataType.VARCHAR, max_length=65_535, is_primary=True, auto_id=False
```

Comment:
I think this needs to be changed to "id" to maintain compatibility with other datasets

Reply:
This does have to be "id". This is the field name of primary key in Milvus. The code will recognize primary key field by checking `is_primary` attribute instead of by its field name. The export & import will use "id" to record and get primary keys.